browser/views.py

```diff
@@ -713,6 +713,60 @@ def delete_investigator_record(request, person_id, research_id):
     return HttpResponseRedirect(reverse('person', args=(person_id,)))
 
 
+@staff_member_required
+def edit_affiliation(request, person_id, affiliation_id):
+    person = get_object_or_404(Person, pk=person_id)
+    affiliation = get_object_or_404(Affiliation, pk=affiliation_id)
+    template = "browser/affiliation_edit.html"
+    institution = None
+
+    form = EditAffiliationForm(initial={'institution': affiliation.institution,
+                                        'position': affiliation.position,
+                                        'institution_id': affiliation.institution.id if affiliation.institution else None})
+    context = {
+        'form': form,
+        'person': person
+    }
+
+    if request.method == 'POST':
+        form = EditAffiliationForm(request.POST, instance=person)
+        if form.is_valid():
+            if form.cleaned_data.get('institution'):
+                # If institution_id is not populated then create the institution before updating affiliation
+                if not form.cleaned_data.get('institution_id'):
+                    # Checking if create institution radio button has been checked
+                    if not form.cleaned_data.get('create_institution'):
+                        messages.add_message(request, messages.ERROR,
+                                             'The above institute does not exist in the database. '
+                                             'Please check the checkbox below to create new institute')
+                        context.update({'form': form})
+                        return render(request, template, context)
+                    else:
+                        institution = Institution.objects.create(
+                            name=form.cleaned_data.get('institution'),
+                            changed_by=request.user
+                        )
+                else:
+                    institution = Institution.objects.get(id=form.cleaned_data.get('institution_id'))
```

Comment:
this seems like the same code you had in your other PR. If so, this should not be duplicated but put in a helper method.